### ISSUE TITLE
Broken APIs after Charmhub cutover

### ISSUE THREAD

**Issue by genet022** [juju/python-libjuju#634]
Previously supported APIs have broke after the switch from Charmstore to Charmhub.
Presumably due to Charmhub not supporting the old APIs.

A simple reproducible example:
```python
>>> model = Model()
>>> do(model.connect_current())
>>> do(model.deploy("cs:~containers/easyrsa"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in do
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.8/dist-packages/juju/model.py", line 1614, in deploy
    res = await self.deploy_types[str(url.schema)].resolve(url, architecture, application_name, channel, series, entity_url)
  File "/usr/local/lib/python3.8/dist-packages/juju/model.py", line 513, in resolve
    app_name = result['Meta']['charm-metadata']['Name']
KeyError: 'charm-metadata'
```
---

A bit more granular:
```
$ curl "https://api.jujucharms.com/charmstore/v5/containers/coredns/meta/any?channel=edge"
{"Code":"not found","Message":"No matching charm or bundle for cs:containers/coredns; for channel: 'edge'"}
```
```
$ curl "https://api.jujucharms.com/charmstore/v5/~containers/coredns/meta/any"
{"Id":"cs:~containers/coredns-20"}
```

**Comment by addyess** [comment]
please note I provided one of these granular details.  Note that in the first example `containers` is missing a tilda (`~`)

`curl`ing the correct endpoint yields a response rather than the 404 Response.  However, this missing metadata still could be an issue

```bash
❯ curl "https://api.jujucharms.com/charmstore/v5/~containers/coredns/meta/any?channel=edge"
{"Id":"cs:~containers/coredns-21"}
```

**Comment by addyess** [comment]
the url to query actually is generated by another pip package [`theblues`](https://github.com/juju/theblues/blob/develop/theblues/charmstore.py#L109)

since the url generated here doesn't have the query parameters `include=id`, this isn't returned to python-libjuju

**Comment by juanmanuel-tirado** [comment]
Fixed after #635
